packages/xo-server/src/api/sm.mjs

```diff
@@ -0,0 +1,14 @@
+export async function get({ id, type }) {
+  return this.getXapi(id).getSmFromSrType(type)
+}
+
+get.description = 'Get SM from SR Type'
+
+get.params = {
+  id: { type: 'string' },
+  type: { type: 'string' },
+}
+
+get.resolve = {
+  sr: ['sr', 'SR', 'view'],
+}
```

Comment:
I don't think it can work.
`*.resolve` take your param # and inject it in your fonction.
`resolve = {injectedParam: ['<paramToResolve>, '<objectType>', '<permission>']}`.

So, I think here, you have to do:
```js
function get({sr}){
 return this.getXapi(sr).getSrmFromSrType(sr.SR_type)
}

get.params = {
 sr: {type: 'string'} // i renamed it sr instead of ID to understand we want an SR ID and not an SM ID
}
get.resolve = {
 sr: ['sr', 'SR', 'view'] 
}
```